code/datums/weather/weather.dm

```diff
@@ -83,18 +79,63 @@
 	/// For barometers to know when the next storm will hit
 	var/next_hit_time = 0
-	/// This causes the weather to only end if forced to
-	var/perpetual = FALSE
-
-/datum/weather/New(z_levels)
+	/// The list of turfs (only /turf/open/ subtypes) that the weather event is being applied to.
+	/// If WEATHER_TURFS or WEATHER_THUNDER weather_flags are not applied this will be an empty list
+	var/list/weather_turfs = list()
+	/// The chance, per tick, a turf will have weather effects applied to it. This is a decimal value, 1.00 = 100%, 0.50 = 50%, etc.
+	/// Recommend setting this low near 0.01 (results in 1 in 100 affected turfs having weather reagents applied per tick)
+	var/turf_weather_chance = 0
+	/// The chance, per tick, a turf will have a thunder strike applied to it. This is a decimal value, 1.00 = 100%, 0.50 = 50%, etc.
+	/// Recommend setting this really low near 0.001 (results in 1 in 1000 affected turfs having thunder strikes applied per tick)
+	var/turf_thunder_chance = THUNDER_CHANCE_AVERAGE // does nothing without the WEATHER_THUNDER weather_flag
+	/// The maximum amount of turfs that can be processed in a single tick regardless of
+	/// the number of turfs determined by turf_weather_chance and turf_thunder_chance
+	/// increasing this too high can result in severe lag so please be careful
+	var/max_turfs_per_tick = 500
+	/// The calculated amount of turfs that get weather effects processed each tick (this gets calculated do not manually set this var)
+	var/weather_turfs_per_tick = 0
+	/// The calculated amount of turfs that get thunder effects processed each tick (this gets calculated do not manually set this var)
+	var/thunder_turfs_per_tick = 0
+	/// Color to apply to thunder while weather is occuring
+	var/thunder_color = null
+
+	/// List of weather bitflags that determines effects (see \code\__DEFINES\weather.dm)
+	var/weather_flags = NONE
+
+	/// List of current mobs being processed by weather
+	var/list/current_mobs = list()
+	/// The weather turf counter to keep track of how many turfs we have processed so far
+	var/turf_iteration = 0
+	/// The weather thunder counter to keep track of how much thunder we have processed so far
+	var/thunder_iteration = 0
+	/// The current section our weather subsystem is processing
+	var/currentpart
+	/// The list of allowed tasks our weather subsystem is allowed to process (determined by weather_flags)
+	var/list/subsystem_tasks = list()
+
+/datum/weather/New(z_levels, area_override, weather_flags_override, thunder_chance_override, datum/reagent/custom_reagent)
 	..()
 	impacted_z_levels = z_levels
+	area_type = area_override || area_type
+	weather_flags = weather_flags_override || weather_flags
+
+	// turf_thunder_chance = thunder_chance_override || turf_thunder_chance
+	// this breaks when thunder_chance_override is 0 (aka FALSE), so we need to null check
+	turf_thunder_chance = !isnull(thunder_chance_override) ? thunder_chance_override : turf_thunder_chance
```

Comment:
```suggestion
	turf_thunder_chance = isnull(thunder_chance_override) ? turf_thunder_chance : thunder_chance_override 
```
something something double negative. that or just like make this an if else that'd also be fine